up42/auth.py

```diff
@@ -210,26 +236,25 @@ def _request(
         Returns:
             The API response.
         """
-        if self.retry:
-            retryer = Retrying(
-                stop=stop_after_attempt(1),  # TODO: Find optimal retry solution
-                wait=wait_fixed(0),
-                retry=(
-                    retry_if_exception_type(requests.exceptions.HTTPError)
-                    | retry_if_exception_type(requests.exceptions.ConnectionError)
-                ),
-                after=self._get_token(),
-            )
-            response = retryer(
-                self._request_helper, request_type, url, data, querystring
-            )
-        else:
-            response = self._request_helper(request_type, url, data, querystring)  # type: ignore
-
         try:
-            response.raise_for_status()
```

Comment:
Yeah the current retry mechanism is not well implemented, was deactivated with `stop_after_attempt(1)`. The self.retry flag is important for some tests where you don't want to see retry behaviour. I think this can be refactored and maybe just combined into your retry mechanism with multiple triggers (or maybe it needs to stay separate if we think the rate limiting should have longer wait times etc). Anyway, would say we merge this and create a techdebt ticket to refactor the retry mechanism + self.retry flag as a whole. Or should look into it right now?

Reply:
Sounds good! I was just afraid maybe this messes some things up. If your comfortable I will then merge and we tackle retrying later.